<a href="https://colab.research.google.com/github/Atulsingh7/DealersData/blob/main/Welcome_To_Colaboratory.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [300]:
import pandas as pd 
import numpy as np 

In [301]:
df_bikes = pd.read_csv('/content/bikes.csv')
df_dealers = pd.read_csv('/content/output.csv')

<ipython-input-301-1386c8c55caf>:2: DtypeWarning: Columns (21,27,28,37,56) have mixed types. Specify dtype option on import or set low_memory=False.
  df_dealers = pd.read_csv('/content/output.csv')


In [302]:
# Removing the nan values of Dealer Code and Cust ID 

df_dealers = df_dealers[df_dealers['Dealer Code'].notna()]
df_bikes = df_bikes[df_bikes['Customer Id'].notna()]

In [303]:
set_dealers = set(df_dealers['Email Id'].tolist())
set_bikes = set(df_bikes['Customer Email Id'].tolist())

In [304]:
# set Intersection of Emails 

intersection_email = set_dealers.intersection(set_bikes)
intersection_email

{'aaa@gmail.com', 'ayyampettai@baluhonda.com', nan, 'viju.ptel@gmail.com'}

In [305]:
dataFrame3 = df_bikes.loc[df_bikes['Customer Email Id'] == 'viju.ptel@gmail.com']
dataFrame3

Region Model Name Type Customer Id Customer First Name  \
84704  Central   CB SHINE   ID   1-2G7HMJ7              PUNAJI   

      Customer Last Name     Customer Type Customer Category  \
84704           KHEDEKAR  First Time Buyer        Individual   

                        Customer Address Line 1 Customer Address Line 2  ...  \
84704  71 , VILL. MACHALPUR BUJURG TAH MAHESWER                     NaN  ...   

      Invoice Number                Reference No Invoice Created Date  \
84704            NaN  VEHINV-MP150001-1617-00602            5/17/2016   

       Financier Name       Frame Number  Engine Number  \
84704          OTHERS  ME4JC651DGT232372  JC65ET0343884   

      Test Ride Vehicle Flag  Purchase type Enquiry Category  \
84704                    NaN        Finance       Individual   

      Number of Invoices  
84704                  1  

[1 rows x 27 columns]

In [306]:
###############################################################################################################

In [307]:
# Changing the Datatype as per need 

df_bikes['Mobile No.'] = df_bikes['Mobile No.'].astype(int)
df_bikes['Mobile No.'] = df_bikes['Mobile No.'].astype(str)

In [308]:
df_bikes['Mobile No.'].shape , df_dealers['Help Line Numbers'].shape

((358703,), (19914,))

In [309]:
# Removing the NULL values 

df_bikes_c = df_bikes[df_bikes['Mobile No.'].notna()]
df_dealers_c = df_dealers[df_dealers['Help Line Numbers'].notna()]


In [310]:
# Making set out of Dataframes 

rough1 =  set(df_dealers['Help Line Numbers'].tolist())
rough2 = set(df_bikes['Mobile No.'].tolist())

In [311]:
# Finding common values 

rough3 = rough1.intersection(rough2)
len(rough3)

87

In [312]:
##############################################################################################

In [313]:
rough3 = list(rough3)

In [314]:
rough3[0]

'9997729105'

In [315]:
# Code to Check the dealers in Customer DB


df_bikes.loc[df_bikes['Mobile No.'] == '9829397355']

Region Model Name Type Customer Id Customer First Name  \
331418  North   CB SHINE  2ID   1-1VI88KG            NARENDRA   
359422  North   CB SHINE  2ID   1-1T1K8SD             KAILASH   

       Customer Last Name     Customer Type Customer Category  \
331418              MEENA  First Time Buyer        Individual   
359422              MEENA  First Time Buyer        Individual   

        Customer Address Line 1 Customer Address Line 2  ... Invoice Number  \
331418    DHUNAI TEHSIL  NADBAI               BHARATPUR  ...            NaN   
359422  VILLAGE - MEENA SEEMALA           TEHSIL SIKRAI  ...            NaN   

                      Reference No Invoice Created Date  Financier Name  \
331418  VEHINV-RJ110001-1617-00049             4/7/2016             NaN   
359422  VEHINV-RJ110001-1617-00006             4/1/2016             NaN   

             Frame Number  Engine Number Test Ride Vehicle Flag  \
331418  ME4JC652BG7143639  JC65E70458067                    NaN   
359422  ME4JC652BG7143605  JC65E70458029                    NaN   

        Purchase type Enquiry Category Number of Invoices  
331418           Cash       Individual                  1  
359422           Cash       Individual                  1  

[2 rows x 27 columns]

In [316]:
# Finding common values 

rough3 = rough1.intersection(rough2)
len(rough3)

87

In [317]:
df_bikes.loc[df_bikes['Mobile No.'] == '7588166780']

Region  Model Name Type Customer Id Customer First Name  \
254045   West    CB SHINE   ID   1-21WVSQ4           RAGHUNATH   
306491   West    CB SHINE   ID   1-1Y24P3Y        PANDHARINATH   
307069   West  DREAM YUGA  2ID   1-1Y0GIPA            RAVINDRA   
350754   West    CB SHINE   ID   1-1U4171T                ASHA   

       Customer Last Name     Customer Type Customer Category  \
254045             MOKASE  First Time Buyer        Individual   
306491            SALUNKE  First Time Buyer        Individual   
307069              KADAM  First Time Buyer        Individual   
350754             PALHAL  First Time Buyer        Individual   

                    Customer Address Line 1 Customer Address Line 2  ...  \
254045  M/POST- PISHOR SHAFEPUR  TL- KANNAD                     NaN  ...   
306491                           A/P SANJUL               PHULAMBRI  ...   
307069                              A/P N-6  AVISHKAR COLONY, CIDCO  ...   
350754          AT/PO- N-9, PRATAPGAD NAGAR                  CIDCO,  ...   

       Invoice Number                Reference No Invoice Created Date  \
254045            NaN  VEHINV-MH060001-1617-00635            4/20/2016   
306491            NaN  VEHINV-MH060001-1617-00437            4/12/2016   
307069            NaN  VEHINV-MH060001-1617-00427            4/12/2016   
350754            NaN  VEHINV-MH060001-1617-00070             4/4/2016   

        Financier Name       Frame Number  Engine Number  \
254045             NaN  ME4JC651AG7283947  JC65E70426151   
306491             NaN  ME4JC651CG7318128  JC65E70477298   
307069             NaN  ME4JC58ACGT247788  JC58ET4247886   
350754             NaN  ME4JC651AGT170085  JC65ET0251008   

       Test Ride Vehicle Flag  Purchase type Enquiry Category  \
254045                    NaN           Cash       Individual   
306491                    NaN           Cash       Individual   
307069                    NaN           Cash       Individual   
350754                    NaN           Cash       Individual   

       Number of Invoices  
254045                  1  
306491                  1  
307069                  1  
350754                  1  

[4 rows x 27 columns]

In [318]:
df_big = pd.concat([dataFrame1,dataFrame2, dataFrame3], ignore_index=True)
##############

for number in rough3:
  dataFrame4 = df_bikes.loc[df_bikes['Mobile No.'] == number]
  df_big = pd.concat([df_big , dataFrame4], ignore_index=True)

df_big


Region      Model Name Type Customer Id Customer First Name  \
0    Central    CD 110 DREAM   ID   1-275DNBN            DEVENDRA   
1    Central      DREAM YUGA  2ID   1-275236V              ASHISH   
2    Central        CB SHINE  7ID   1-2610VGN               BHANU   
3    Central      DREAM YUGA  2ID   1-260VPKR              NEERAJ   
4      South        ACTIVA I  5ID   1-2NCUYND       MOHAMED NASAR   
..       ...             ...  ...         ...                 ...   
346     West        CB SHINE  2ID   1-1X9KCUQ            DNYANDEV   
347     West        CB SHINE  2ID   1-1XF0NLN              MAROTI   
348     West        CB SHINE   ID   1-1XFB36Y              SONAJI   
349     West  CB125 SHINE SP  3ID   1-1XCEBG9               SURAJ   
350     West            LIVO   ID   1-1XA8VJO              GOVIND   

    Customer Last Name     Customer Type Customer Category  \
0               SHARMA  First Time Buyer        Individual   
1               TIWARI  First Time Buyer        Individual   
2                TOMAR  First Time Buyer        Individual   
3                  NaN  First Time Buyer        Individual   
4                    S  First Time Buyer        Individual   
..                 ...               ...               ...   
346            KHARATE  First Time Buyer        Individual   
347             SHINDE  First Time Buyer        Individual   
348             DHUMAL  First Time Buyer        Individual   
349             CHAVAN  First Time Buyer        Individual   
350           KHANDARE  First Time Buyer        Individual   

                    Customer Address Line 1      Customer Address Line 2  ...  \
0                          KUTHIYANA, AMBAH                          NaN  ...   
1                               GORMI BHIND                          NaN  ...   
2                       PALI VIJAYGAD PORSA                          NaN  ...   
3                          GRAM- KUMAR PURA                          NaN  ...   
4    NO: 96A,NORTH STREET,CHAKKARAPALLI(PO)                 PAPANASAM-TK  ...   
..                                      ...                          ...  ...   
346                           A/P - KAVATHA     TQ- BASMAT DIST- HINGOLI  ...   
347                                A/P- GUR     TQ- ARDHAPUR DIST-NANDED  ...   
348                            A/P- BASMAT.       TQ-BASMAT DIST-HINGOLI  ...   
349                           A/P- PARBHANI            TQ-DIST- PARBHANI  ...   
350                         A/P- JAVALA (P)  TQ- KALAMNURI DIST- HINGOLI  ...   

    Invoice Number                Reference No Invoice Created Date  \
0              NaN  VEHINV-MP320001-1617-00270            4/30/2016   
1              NaN  VEHINV-MP320001-1617-00269            4/30/2016   
2              NaN  VEHINV-MP320001-1617-00265            4/28/2016   
3              NaN  VEHINV-MP320001-1617-00264            4/28/2016   
4              NaN  VEHINV-TN140002-1617-00534            5/30/2016   
..             ...                         ...                  ...   
346            NaN  VEHINV-MH380001-1617-00035            4/11/2016   
347            NaN  VEHINV-MH380001-1617-00041            4/11/2016   
348            NaN  VEHINV-MH380001-1617-00042            4/11/2016   
349            NaN  VEHINV-MH380001-1617-00038            4/11/2016   
350            NaN  VEHINV-MH380001-1617-00036            4/11/2016   

       Financier Name       Frame Number  Engine Number  \
0                 NaN  ME4JC671KF8141577  JC67E80141535   
1                 NaN  ME4JC58ADGT253557  JC58ET4253746   
2                 NaN  ME4JC36NJE7086103  JC36E73687673   
3                 NaN  ME4JC58ADGT253931  JC58ET4254074   
4              OTHERS  ME4JF487DG8021055  JF48E81089062   
..                ...                ...            ...   
346  Shri Ram Finance  ME4JC652CG7148114  JC65E70471221   
347               NaN  ME4JC652BG7144871  JC65E70462037   
348  Shri Ram Finance  ME4JC651BG7300200  JC65E70451892   
349             

In [319]:
############################################################################################################

In [320]:
############################################################################################################

In [321]:
# Removing the nan values of Dealer Code and Cust ID 
df_bikes_orig = df_bikes
df_dealers = df_dealers[df_dealers['Dealer Code'].notna()]
df_bikes = df_bikes[df_bikes['Phone No. Office'].notna()]

In [322]:
# Changing the Datatype as per need 

df_bikes['Phone No. Office'] = df_bikes['Phone No. Office'].astype(int)
df_bikes['Phone No. Office'] = df_bikes['Phone No. Office'].astype(str)

<ipython-input-322-6f3e032c57ff>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_bikes['Phone No. Office'] = df_bikes['Phone No. Office'].astype(int)
<ipython-input-322-6f3e032c57ff>:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_bikes['Phone No. Office'] = df_bikes['Phone No. Office'].astype(str)


In [323]:
df_bikes.shape , df_dealers.shape

((11024, 27), (19914, 81))

In [324]:
set_dealers = set(df_dealers['Help Line Numbers'].tolist())
set_bikes = set(df_bikes['Phone No. Office'].tolist())

In [325]:
rough4 = set_dealers.intersection(set_bikes)

In [326]:
rough4

{'123456789', '8814980327', '8888819557'}

In [327]:
############################################################################################################

In [328]:
# Removing the nan values of Dealer Code and Cust ID 

df_dealers = df_dealers[df_dealers['Dealer Code'].notna()]
df_bikes = df_bikes[df_bikes['Phone No. Residence'].notna()]

In [329]:
# Changing the Datatype as per need 

df_bikes['Phone No. Residence'] = df_bikes['Phone No. Residence'].astype(int)
df_bikes['Phone No. Residence'] = df_bikes['Phone No. Residence'].astype(str)

<ipython-input-329-6ebc17c5dba5>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_bikes['Phone No. Residence'] = df_bikes['Phone No. Residence'].astype(int)
<ipython-input-329-6ebc17c5dba5>:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_bikes['Phone No. Residence'] = df_bikes['Phone No. Residence'].astype(str)


In [330]:
df_bikes.shape , df_dealers.shape

((1932, 27), (19914, 81))

In [331]:
set_dealers = set(df_dealers['Help Line Numbers'].tolist())
set_bikes = set(df_bikes['Phone No. Residence'].tolist())

In [332]:
rough5 = set_dealers.intersection(set_bikes)
rough5

{'8275302645'}

In [333]:
############################################################################################################

In [334]:
set_dealers = set(df_dealers['Customer Care No'].tolist())
set_bikes = set(df_bikes['Phone No. Residence'].tolist())

In [335]:
rough6 = set_dealers.intersection(set_bikes)
rough6


set()

In [336]:
############################################################################################################

In [337]:
set_dealers = set(df_dealers['Service Contact Numbers'].tolist())
set_bikes = set(df_bikes['Mobile No.'].tolist())

In [338]:
rough7 = set_dealers.intersection(set_bikes)
rough7

{'8059900601'}

In [339]:
set_dealers = set(df_dealers['Service Contact Numbers'].tolist())
set_bikes = set(df_bikes['Phone No. Residence'].tolist())
rough8 = set_dealers.intersection(set_bikes)
rough8

set()

In [340]:
len(rough1),len(rough2),len(rough3)

(3031, 345628, 87)

In [341]:
###############################################################################################################

In [342]:
len(rough4),len(rough5),len(rough6),len(rough7)


(3, 1, 0, 1)

In [348]:
rough6 = rough4.union(rough5)
roughFinal = rough6.union(rough7)

len(roughFinal)

5

In [ ]:
351

In [349]:
for number in roughFinal:
  dataFrame4 = df_bikes_orig.loc[df_bikes_orig['Mobile No.'] == number]
  df_big = pd.concat([df_big , dataFrame4], ignore_index=True)

df_big

Region      Model Name Type Customer Id Customer First Name  \
0    Central    CD 110 DREAM   ID   1-275DNBN            DEVENDRA   
1    Central      DREAM YUGA  2ID   1-275236V              ASHISH   
2    Central        CB SHINE  7ID   1-2610VGN               BHANU   
3    Central      DREAM YUGA  2ID   1-260VPKR              NEERAJ   
4      South        ACTIVA I  5ID   1-2NCUYND       MOHAMED NASAR   
..       ...             ...  ...         ...                 ...   
394    North            LIVO   ID   1-1TYY2HE        RAKESH KUMAR   
395     West        CB SHINE   ID   1-2F3VVOC               VIJAY   
396     West        CB SHINE   ID   1-280FP8T           BASAVARAJ   
397     West  CB125 SHINE SP   ID   1-23HKM9Q           ABHAYSING   
398     West        CB SHINE   ID   1-1ZTSNMI             RAMAJAN   

       Customer Last Name     Customer Type Customer Category  \
0                  SHARMA  First Time Buyer        Individual   
1                  TIWARI  First Time Buyer        Individual   
2                   TOMAR  First Time Buyer        Individual   
3                     NaN  First Time Buyer        Individual   
4                       S  First Time Buyer        Individual   
..                    ...               ...               ...   
394  S/O RAJESHWAR MANDAL  First Time Buyer        Individual   
395                SAWANT  First Time Buyer        Individual   
396              GONDHALI  First Time Buyer        Individual   
397               LANDAGE  First Time Buyer        Individual   
398               PAKHALI  First Time Buyer        Individual   

                        Customer Address Line 1  \
0                              KUTHIYANA, AMBAH   
1                                   GORMI BHIND   
2                           PALI VIJAYGAD PORSA   
3                              GRAM- KUMAR PURA   
4        NO: 96A,NORTH STREET,CHAKKARAPALLI(PO)   
..                                          ...   
394  C/O MODERN AUTO MOTIVES,VILL-CHATTAR PURA,   
395                            A/P NIGADI KHURD   
396                               A/P BIJJARAGI   
397                                  A/P YELAVI   
398              A/P NEW JANAVAR BAZAR SAMBHAJI   

             Customer Address Line 2  ... Invoice Number  \
0                                NaN  ...            NaN   
1                                NaN  ...            NaN   
2                                NaN  ...            NaN   
3                                NaN  ...            NaN   
4                       PAPANASAM-TK  ...            NaN   
..                               ...  ...            ...   
394                 MANDI GOBINDGARH  ...            NaN   
395                         TAL-JATH  ...            NaN   
396                TQ & DT : BIJAPUR  ...            NaN   
397                         TAL-JATH  ...            NaN   
398  NAGAR SATARA ROAD JATH TAL-JATH  ...            NaN   

                   Reference No Invoice Created Date  Financier Name  \
0    VEHINV-MP320001-1617-00270            4/30/2016             NaN   
1    VEHINV-MP320001-1617-00269            4/30/2016             NaN   
2    VEHINV-MP320001-1617-00265            4/28/2016             NaN   
3    VEHINV-MP320001-1617-00264            4/28/2016             NaN   
4    VEHINV-TN140002-1617-00534            5/30/2016          OTHERS   
..                          ...                  ...             ...   
394  VEHINV-PB140002-1617-00019             4/4/2016          OTHERS   
395  VEHINV-MH100001-1617-01542            5/15/2016             NaN   
396  VEHINV-MH100001-1617-00972             5/2/2016             NaN   
397  VEHINV-MH100001-1617-00737            4/23/2016             NaN   
398  VEHINV-MH100001-1617-00289            4/16/2016             NaN   

          Frame Number  Engine Number Test Ride Vehicle Flag  Purchase type  \
0    ME4JC671KF8141577  JC67E80141535                    NaN           Cash   
1    ME4JC58ADGT253557  JC58ET4253746  

SO WE HAVE 350 DIFFERENT USERS WITH UNIQUE CUSTOMER ID


In [351]:
gb1 = df_big.groupby('Customer Id')
gb1.first()


Region      Model Name Type        Customer First Name  \
Customer Id                                                          
1-1QJ9YVO     West      ACTIVA 125   ID                     TAUSIF   
1-1SMYNR1     West        CB SHINE   ID                    PRADEEP   
1-1SNSBEL     West      ACTIVA 125   ID  YASH MOTORS(NASHTE HONDA)   
1-1SQXI69     West  CB125 SHINE SP  3ID                      DILIP   
1-1T0ZBFP    North    CD 110 DREAM   ID                   AMARJEET   
...            ...             ...  ...                        ...   
1-2O3SY9B     West      CB UNICORN  7ID                    SANTOSH   
1-2O4HG4B    South      DREAM YUGA  2ID                 MAHALINGAM   
1-2O4YDEO    South            LIVO   ID           DAKSHINA MOORTHY   
1-2O58PN9     West  CB125 SHINE SP   ID           SHAIKH ANISODDIN   
1-2OBHF1L     West      ACTIVA 125   ID               HARSHWARDHAN   

                                Customer Last Name     Customer Type  \
Customer Id                                                            
1-1QJ9YVO                                    PATEL  First Time Buyer   
1-1SMYNR1                                     BELE  First Time Buyer   
1-1SNSBEL                                     None  First Time Buyer   
1-1SQXI69    PATIL M S SAIPOINT AUTOMOBILE PVT LTD  First Time Buyer   
1-1T0ZBFP                                   CHADHA  First Time Buyer   
...                                            ...               ...   
1-2O3SY9B                                   CHAVAN  First Time Buyer   
1-2O4HG4B                                        M  First Time Buyer   
1-2O4YDEO                                        P  First Time Buyer   
1-2O58PN9                       SHAIKH RASHIYODDIN  First Time Buyer   
1-2OBHF1L                                 GAJBHIYE  First Time Buyer   

            Customer Category  \
Customer Id                     
1-1QJ9YVO          Individual   
1-1SMYNR1          Individual   
1-1SNSBEL          Individual   
1-1SQXI69          Individual   
1-1T0ZBFP          Individual   
...                       ...   
1-2O3SY9B          Individual   
1-2O4HG4B          Individual   
1-2O4YDEO          Individual   
1-2O58PN9          Individual   
1-2OBHF1L          Individual   

                                       Customer Address Line 1  \
Customer Id                                                      
1-1QJ9YVO                            VIDYAMANDIR PANDHARE PANI   
1-1SMYNR1                               CHANODANADRA PO-HIWARA   
1-1SNSBEL               A/P- OPP.SBI BANK,GANDHI CHOWK AKLUJ*-   
1-1SQXI69    SAVLARAM COM NR V PAWAN PUTRA RESIDENCY OPP HO...   
1-1T0ZBFP                        # 178/3, KRISHNA COLONY,LADWA   
...                                                        ...   
1-2O3SY9B                           AT/PO- BUDHWAR PETH BASMAT   
1-2O4HG4B                                       DEVANESA NAGAR   
1-2O4YDEO                                    ANNA CROSS STREET   
1-2O58PN9                                         AT/PO-PATHRI   
1-2OBHF1L                         H/N-221 W/NO-03 SANGAM ROAD.   

                 Customer Address Line 2         Locality  ... Invoice Number  \
Customer Id                                                ...                  
1-1QJ9YVO     MALAI WADA   TAL-SHAHUWADI             None  ...            NaN   
1-1SMYNR1         HIWARI PIPLA TQ. UMRED             None  ...            NaN   
1-1SNSBEL     TAL-MALSHIRAS DIST-SOLAPUR             None  ...            NaN   
1-1SQXI69            HOTEL SAI POOJA OLD             None  ...            NaN   
1-1T0ZBFP                  TEH: THANESAR             None  ...            NaN   
...                                  ...              ...  ...            ...   
1-2O3SY9B        TQ-BASMATH DIST-HINGOLI   Basmath-431512  ...            NaN   
1-2O4HG4B         PEERKANKARANAI CHENNAI             None  ...            NaN   
1-2O4YDEO      NEW PERUNGALATHUR CHENNAI             None  ...   

In [364]:

count = df_big['Mobile No.'].value_counts()
count

123456789       86
8007999326      25
9997729105      18
9551024000      15
9673098371      14
                ..
8129460555       1
9174094021.0     1
9942682865.0     1
9043648876.0     1
8059900601       1
Name: Mobile No., Length: 120, dtype: int64

In [368]:
df_big.loc[df_big['Mobile No.'] == '8007999326']

Region      Model Name Type Customer Id Customer First Name  \
297   West      ACTIVA 125   ID   1-2OBHF1L        HARSHWARDHAN   
298   West        CB SHINE   ID   1-2LSXF89               VIJAY   
299   West         AVIATOR  4ID   1-2IQUGE2                ARUN   
300   West      DREAM YUGA  2ID   1-2IMIZ6H              SATISH   
301   West        CB SHINE   ID   1-2GM7U3T            SURENDRA   
302   West      DREAM YUGA  2ID   1-2FFBA86             BABULAL   
303   West        CB SHINE   ID   1-2CM1B8Z         CHANDRAKALA   
304   West        ACTIVA I  5ID   1-2939F99             RAJRANI   
305   West        CB SHINE   ID   1-293R3NT               DIPAK   
306   West            LIVO  3ID   1-2940MOF           RAMCHARAN   
307   West        CB SHINE   ID   1-268A2QG               AKASH   
308   West            LIVO   ID   1-24DFGZW              NAGESH   
309   West      ACTIVA 125   ID   1-238JJS7             ASHWINI   
310   West        CB SHINE  2ID   1-23732UX            RAJENDRA   
311   West  CB HORNET 160R   ID   1-21U46JU         HEMANTKUMAR   
312   West      DREAM YUGA  2ID   1-21WDG2E         KAMTAPRASAD   
313   West        CB SHINE   ID   1-1YN2EHY          RAMKRISHNA   
314   West        CB SHINE   ID   1-1YGM3IK               DILIP   
315   West            LIVO   ID   1-1YGA4ZS               ASHOK   
316   West      DREAM YUGA  2ID   1-1XCRRQF                ATUL   
317   West        CB SHINE   ID   1-1UQV0YH              PRAMOD   
318   West      DREAM YUGA  2ID   1-1U975LX            DIGAMBER   
319   West      ACTIVA 125   ID   1-1U7ZG4D              KAVITA   
320   West        CB SHINE   ID   1-1U88CKE                RAMU   
321   West        CB SHINE   ID   1-1SMYNR1             PRADEEP   

    Customer Last Name     Customer Type Customer Category  \
297           GAJBHIYE  First Time Buyer        Individual   
298          RANGANKAR  First Time Buyer        Individual   
299             BHOYAR  First Time Buyer        Individual   
300            MESHRAM  First Time Buyer        Individual   
301           GAJBHIYE  First Time Buyer        Individual   
302          KAITHAVAS  First Time Buyer        Individual   
303             PAHADE  First Time Buyer        Individual   
304               SONI  First Time Buyer        Individual   
305            KANGALE  First Time Buyer        Individual   
306            KAIKADE  First Time Buyer        Individual   
307              BAGDE  First Time Buyer        Individual   
308           SONTAKKE  First Time Buyer        Individual   
309              DHOLE  First Time Buyer        Individual   
310              YADAV  First Time Buyer        Individual   
311             SAYAME  First Time Buyer        Individual   
312           KAITHWAS  First Time Buyer        Individual   
313             BOPCHE  First Time Buyer        Individual   
314         SHRIVASTAV  First Time Buyer        Individual   
315            DAWANDE  First Time Buyer        Individual   
316             KHADSE  First Time Buyer        Individual   
317          GANDHEWAR  First Time Buyer        Individual   
318           HIREKHAN  First Time Buyer        Individual   
319              INGLE  First Time Buyer        Individual   
320             MANKAR  First Time Buyer        Individual   
321               BELE  First Time Buyer        Individual   

                    Customer Address Line 1  \
297            H/N-221 W/NO-03 SANGAM ROAD.   
298                 H/NO-64 W/NO-02 SINGORI   
299                   W/N-01 AT-PO-DAHEGAON   
300    54-1 SHAM MANGAL KARYALAY KADIL BHAG   
301                     B/H-PRIMARY SCHOOL.   
302            Q/NO-70/2 W/NO-4 AT-CHABAPUR   
303                  RETA CHECK POST W/N-02   
304       W NO. 5  OM NAGAR  POTA CHANKAPUR   
305                BHANEGAON  22 NAVIN BINA   
306                  73/2  CHINCHOLI  K  P2   
307                           NR- ANGANWADI   
308                    W/N-03 KAMPTEE ROAD.   
309  PLOT NO-217 W/N-04 ANNAMOD KHAPE

In [347]:
# This code helps in deleting entries of the DataFrame

'''df = df_bikes

for number in rough3:
  index_names = df[ df['Mobile No.'] == number ].index
  df.drop(index_names, inplace = True)

df.shape 
'''

"df = df_bikes\n\nfor number in rough3:\n  index_names = df[ df['Mobile No.'] == number ].index\n  df.drop(index_names, inplace = True)\n\ndf.shape \n"